pkg/server/api/agent/v1/service.go

```diff
@@ -88,13 +104,45 @@ func (s *Service) CreateJoinToken(ctx context.Context, req *agent.CreateJoinToke
 		},
 	})
 	if err != nil {
-		log.WithError(err).Error("Failed to register token")
-		return nil, status.Errorf(codes.Internal, "Failed to register token: %v", err)
+		log.WithError(err).Error("Failed to create token")
+		return nil, status.Errorf(codes.Internal, "Failed to create token: %v", err)
+	}
+
+	if req.AgentId != nil {
+		err = s.createJoinTokenRegistrationEntry(ctx, req.Token, req.AgentId)
+		if err != nil {
+			return nil, status.Errorf(codes.InvalidArgument, "Failed to create registration entry: %v", err)
```

Comment:
Not every failure is due to InvalidArgument. I'd suggest that the validation above handle the invalid arguments since those can all be detected before creating the join token. Either way, `createJoinTokenRegistrationEntry` should probably return errors that are created via `status.Error(f)` so they have the proper code attached.